src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -97,213 +92,409 @@ Position const SpawnBat[6] =
     {
-        _scheduler.SetValidator([this]
+        // if this is a summon of Jeklik, it is in boss mode
+        if
+        (
+            me->GetEntry() == NPC_BATRIDER &&
+            me->IsSummon() &&
+            me->ToTempSummon() &&
+            me->ToTempSummon()->GetSummoner() &&
+            me->ToTempSummon()->GetSummoner()->GetEntry() == NPC_PRIESTESS_JEKLIK
+        )
         {
-            return !me->HasUnitState(UNIT_STATE_CASTING);
-        });
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BATRIDER_MODE_BOSS");
+            _mode = BATRIDER_MODE_BOSS;
+
+            // make the bat rider unattackable
+            me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+            // keep the bat from attacking players directly
+            me->SetReactState(REACT_PASSIVE);
+
+            // make the bat rider move the correct speed
+            me->SetSpeed(MOVE_WALK, 5.0f, true);
+
+            // start the flight loop
+            me->SetCanFly(true);
+            me->SetDisableGravity(true);
+            me->GetMotionMaster()->MoveSplinePath(PATH_BATRIDER_LOOP);
+        }
+        // otherwise, trash mode
+        else
+        {
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BATRIDER_MODE_TRASH");
+            me->SetReactState(REACT_DEFENSIVE);
+            _mode = BATRIDER_MODE_TRASH;
+
+            // don't interrupt casting
+            _scheduler.SetValidator([this]
+            {
+                return !me->HasUnitState(UNIT_STATE_CASTING);
+            });
+        }
     }
 
     void Reset() override
     {
+        CreatureAI::Reset();
+
         _scheduler.CancelAll();
-        me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-        me->SetHover(true);
-        me->SetDisableGravity(true);
-        me->AddUnitState(UNIT_STATE_ROOT);
+
+        if (_mode == BATRIDER_MODE_BOSS)
+        {
+            me->GetMotionMaster()->Clear();
+        }
+        else if (_mode == BATRIDER_MODE_TRASH)
+        {
+            // apply the Thrash (8876) spell to the bat rider (passive ability)
+            me->CastSpell(me, SPELL_BATRIDER_PASSIVE_THRASH);
+        }
     }
 
-    void JustEngagedWith(Unit* /*who*/) override
+    void JustEngagedWith(Unit* who) override
     {
-        _scheduler.Schedule(2s, [this](TaskContext context)
+        CreatureAI::JustEngagedWith(who);
+
+        if (_mode == BATRIDER_MODE_BOSS)
         {
-            DoCastRandomTarget(SPELL_THROW_LIQUID_FIRE);
-            context.Repeat(7s);
-        });
+            // throw bomb
+            _scheduler.Schedule(2s, [this](TaskContext context)
+            {
+                DoCastRandomTarget(SPELL_BATRIDER_THROW_LIQUID_FIRE);
+                context.Repeat(8s);
+            });
+        }
+        else if (_mode == BATRIDER_MODE_TRASH)
+        {
+            // demo shout
```

Comment:
The styling of this scheduler block is quite different from Jeklik's scheduling. This block has spacing between each spell, plus lowercase spell name comments. I personally prefer Jeklik's styling.

Reply:
Fixed.